### PR TITLE
Gyro Join/Exit

### PR DESCRIPTION
**Pull request by lgahdl**
Using the WeightedJoin and WeightedExit, but filtering unsupported join/exit kinds in the validateInputs method.
Adding tests.

### CODE REVIEW

File: src/entities/exit/poolExit.ts [hunk 1]
```diff
@@ -18,6 +18,9 @@ export class PoolExit {
     constructor(config?: ExitConfig) {
         const { customPoolExits } = config || {};
         this.poolExits = {
+            GYRO2: new WeightedExit(),
```
Comment (johngrantuk) [inline src/entities/exit/poolExit.ts:21]: Could we add a comment here explaining what Exits Gyro supports

File: src/entities/join/poolJoin.ts [hunk 2]
```diff
@@ -18,10 +18,13 @@ export class PoolJoin {
     constructor(config?: JoinConfig) {
         const { customPoolJoins } = config || {};
         this.poolJoins = {
-            WEIGHTED: new WeightedJoin(),
             // PHANTOM_STABLE === ComposableStables in API
-            PHANTOM_STABLE: new ComposableStableJoin(),
             // custom pool Joins take precedence over base Joins
+            PHANTOM_STABLE: new ComposableStableJoin(),
+            GYRO2: new WeightedJoin(),
```
Comment (johngrantuk) [inline src/entities/join/poolJoin.ts:24]: Could we add a comment here explaining what Joins Gyro supports

**Review by johngrantuk — CHANGES_REQUESTED** [review]
Nice to reuse the Weighted logic!
Can we add integration tests for the other gyro pool types please.

**Comment by lgahdl** [comment]
> Nice to reuse the Weighted logic! Can we add integration tests for the other gyro pool types please.

Couldn't add tests with native assets because there are no GyroE V1, Gyro2 and Gyro3 pools with wrapped native assets in any network :/

File: test/gyro3Exit.integration.test.ts [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+});
+
+/*********************** Mock To Represent API Requirements **********************/
+
+export class MockApi {
+    public async getPool(id: Hex): Promise<PoolStateInput> {
+        return {
+            id,
+            address: getPoolAddress(id) as Address,
+            type: 'GYRO2',
```
Comment (johngrantuk) [inline test/gyro3Exit.integration.test.ts:169]: Needs to be changed to test properly

File: test/gyro3Join.integration.test.ts [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+});
+
+/*********************** Mock To Represent API Requirements **********************/
+
+export class MockApi {
+    public async getPool(id: Hex): Promise<PoolStateInput> {
+        return {
+            id,
+            address: getPoolAddress(id) as Address,
+            type: 'GYRO2',
```
Comment (johngrantuk) [inline test/gyro3Join.integration.test.ts:186]: Needs to be changed to test properly

**Comment by johngrantuk** [comment]
@lgahdl - notice changes made to buildCall outputs [here](https://github.com/balancer/b-sdk/pull/123). You'll need to also update this PR.

**Comment by lgahdl** [comment]
> @lgahdl - notice changes made to buildCall outputs [here](https://github.com/balancer/b-sdk/pull/123). You'll need to also update this PR.

I merged the branch, everything working fine, the Gyro pools are using the WeightedJoin and WeightedExit.

File: test/gyro3Join.integration.test.ts (hunk 4 above)
Comment (lgahdl) [reply test/gyro3Join.integration.test.ts]: No, just fixed it, thanks!

File: test/gyro3Exit.integration.test.ts (hunk 3 above)
Comment (lgahdl) [reply test/gyro3Exit.integration.test.ts]: No, sorry, didn't see that

**Review by johngrantuk — APPROVED** [review]
LGTM!

**Pull request merged**